services/src/main/java/org/apache/unomi/services/impl/rules/RulesServiceImpl.java

```diff
@@ -277,11 +277,26 @@ public void refreshRules() {
 
     private List<Rule> getAllRules() {
         List<Rule> rules = persistenceService.getAllItems(Rule.class, 0, -1, "priority").getList();
+        List<Rule> rulesToDisable = new ArrayList<>();
+        List<Rule> enabledRules = new ArrayList<>();
         for (Rule rule : rules) {
-            ParserHelper.resolveConditionType(definitionsService, rule.getCondition(), "rule " + rule.getItemId());
-            ParserHelper.resolveActionTypes(definitionsService, rule);
+            // Check rule integrity
+            boolean isValid = ParserHelper.resolveConditionType(definitionsService, rule.getCondition(), "rule " + rule.getItemId());
+            isValid = isValid && ParserHelper.resolveActionTypes(definitionsService, rule);
+            // exclude enabled invalid rules
+            if (isValid) {
+                enabledRules.add(rule);
+            } else if (rule.getMetadata().isEnabled()) {
+                rulesToDisable.add(rule);
+            }
         }
-        return rules;
+        // Disable invalid rules and store it.
+        rulesToDisable.forEach(rule -> {
```

Comment:
I wonder if we should put a setting around this ?

Reply:
Nice catch, get all rules should return always ALL rules